index.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const MergeTrees = require('broccoli-merge-trees');
+const Funnel = require('broccoli-funnel');
+const buildTree = require('./lib/build-tree');
+const FilterLiterals = require('./lib/filter-literals');
+const ReduceLiterals = require('./lib/reduce-literals');
+const SplitTranslations = require('./lib/split-translations');
+const WatchedDir = require('broccoli-source').WatchedDir;
+
+module.exports = {
+	name: require('./package').name,
+
+	isDevelopingAddon() {
+		return true;
+	},
+
+	included() {
+		this._super.included.apply(this, arguments);
+
+		this.app = this._findHost();
+		this.options = Object.assign({
+			files: {},
+			addons: {},
+			translationsDir: 'translations'
+		}, this.app.options['ember-cli-intl-shake']);
+	},
+
+	postprocessTree(type, tree) {
+		if (type !== 'all') {
+			return tree;
+		}
+
+		const trees = buildTree(this.app, this.treeGenerator, this.options);
+
+		if (this.options.directories && this.options.directories.include) {
+			trees.push(...this.options.directories.include.map((directory) =>
+				new Funnel(new WatchedDir(directory), {
+					destDir: this.app.name,
+					annotation: `ember-cli-intl-shake:build-tree:${directory}`
+				})));
+		}
+
+		let intlTree = new Funnel(new MergeTrees(trees, { overwrite: true }), { include: ['**/*.js', '**/*.json', '**/*.hbs'] });
```

Comment:
You shouldn't add the files extensions directly because an Ember project can use TypeScript or different build system to allow other file extensions.
You can retrieve the extensions using something [like this](https://github.com/BBVAEngineering/ember-cli-custom-addons/blob/master/index.js#L84):

```js
return this.registry.extensionsForType('template').map((extension) => `**/*/template.${extension}`);
```

Reply:
Done. Finally add new option to add new extensions and filters.